tiledb/sm/query/readers/sparse_index_reader_base.h

```diff
@@ -99,6 +99,122 @@ class FragIdx {
   uint64_t cell_idx_;
 };
 
+class MemoryBudget {
+ public:
+  /* ********************************* */
+  /*     CONSTRUCTORS & DESTRUCTORS    */
+  /* ********************************* */
+
+  MemoryBudget() = delete;
+
+  MemoryBudget(Config& config, std::string reader_string) {
+    refresh_config(config, reader_string);
+  }
+
+  DISABLE_COPY_AND_COPY_ASSIGN(MemoryBudget);
+  DISABLE_MOVE_AND_MOVE_ASSIGN(MemoryBudget);
+
+  /* ********************************* */
+  /*          PUBLIC METHODS           */
+  /* ********************************* */
+
+  /**
+   * Refresh the configuration.
+   *
+   * @param config Config object.
+   * @param reader_string String to identify the reader settings to load.
+   */
+  void refresh_config(Config& config, std::string reader_string) {
```

Comment:
Suggest making this private (or moving contents into the constructor). I believe this is only strictly needed in the `set_config` call, but there you could reassign.

Reply:
This is needed for the cloud codepath where the config changes after the query is created :(